File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/command/execution/CommandPolicyContext.java
package top.likoslupus.cellulosesz.api.command.execution;

import top.likoslupus.cellulosesz.api.text.LocalizedMessage;

import java.util.Optional;
import java.util.UUID;

public interface CommandPolicyContext {

    String invokedLabel();

    String canonicalRoot();

    boolean player();

    Optional<UUID> playerUuid();

    Optional<String> playerName();

    boolean hasPermission(String permission);

    String auditSummary();

    void reply(LocalizedMessage message);

    void error(LocalizedMessage message);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/command/execution/CommandPolicyContext.kt
package top.likoslupus.cellulosesz.api.command.execution

import top.likoslupus.cellulosesz.api.text.LocalizedMessage
import java.util.*

public interface CommandPolicyContext {

    public fun invokedLabel(): String

    public fun canonicalRoot(): String

    public fun player(): Boolean

    public fun playerUuid(): UUID?

    public fun playerName(): String?

    public fun hasPermission(permission: String): Boolean

    public fun auditSummary(): String

    public fun reply(message: LocalizedMessage)

    public fun error(message: LocalizedMessage)

}
